src/ray/core_worker/core_worker.cc

```diff
@@ -764,12 +784,11 @@ void CoreWorker::HandleAssignTask(const rpc::AssignTaskRequest &request,
   }
 }
 
-void CoreWorker::HandleDirectActorAssignTask(
-    const rpc::DirectActorAssignTaskRequest &request,
-    rpc::DirectActorAssignTaskReply *reply, rpc::SendReplyCallback send_reply_callback) {
+void CoreWorker::HandlePushTask(const rpc::PushTaskRequest &request,
```

Comment:
I'm not sure why PushTask gets so much hate, it makes perfect sense to me.

Reply:
I think the real answer here is that AssignTask is deprecated and we should remove it once direct task calls is complete. That way, there will be no confusion and the semantics will be accurate.

Reply:
Haha I don't mind PushTask. But we should make it clear that it's for actor tasks (`PushTaskImmediate` was not that clear to me).

Reply:
It's for both now though.

Reply:
Ah I see, I got confused because it dispatches to `direct_actor_task_receiver_`, but I guess that class handles both actor and normal task submission?

Reply:
I got confused here too... if it's used for both actor and non-actor tasks, could we rename the `direct_actor_task_receiver_` to something more accurate, e.g. `direct_receiver_`?